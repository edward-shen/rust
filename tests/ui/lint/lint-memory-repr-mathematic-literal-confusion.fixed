// run-rustfix
fn main() {
    let _: i32 = -0b1000_0000_0000_0000_0000_0000_0000_0000i32;
    //~^ ERROR literal out of range
    //~| HELP literals in binary or hexidecimal
    let _: i32 = -0b1000_0000_0000_0000_0000_0000_0000_0000i32;
    let _: i32 = -0b1000_0000_0000_0000_0000_0000_0000_0000;
    //~^ ERROR literal out of range
    //~| HELP literals in binary or hexidecimal
    let _: i32 = -0b1000_0000_0000_0000_0000_0000_0000_0000;
}
