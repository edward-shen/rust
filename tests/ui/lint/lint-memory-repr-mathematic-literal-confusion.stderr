error: literal out of range for `i32`
  --> $DIR/lint-memory-repr-mathematic-literal-confusion.rs:3:18
   |
LL |     let _: i32 = 0b1000_0000_0000_0000_0000_0000_0000_0000i32;
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the literal `0b1000_0000_0000_0000_0000_0000_0000_0000i32` (decimal `2147483648`) does not fit into the type `i32` and will become `-2147483648i32`
   = note: `#[deny(overflowing_literals)]` on by default
help: literals in binary or hexidecimal notation represent mathematical values, not memory layouts
   |
LL |     let _: i32 = -0b1000_0000_0000_0000_0000_0000_0000_0000i32;
   |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: literal out of range for `i32`
  --> $DIR/lint-memory-repr-mathematic-literal-confusion.rs:7:18
   |
LL |     let _: i32 = 0b1000_0000_0000_0000_0000_0000_0000_0000;
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: the literal `0b1000_0000_0000_0000_0000_0000_0000_0000` (decimal `2147483648`) does not fit into the type `i32` and will become `-2147483648i32`
help: literals in binary or hexidecimal notation represent mathematical values, not memory layouts
   |
LL |     let _: i32 = -0b1000_0000_0000_0000_0000_0000_0000_0000;
   |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

error: aborting due to 2 previous errors

